src/services/deployment.ts

```diff
@@ -162,25 +162,31 @@ export const gitlabAPiDeploymentToCompassDataProviderDeploymentEvent = (
   projectName: string,
   environmentTier: EnvironmentTier,
 ): DataProviderDeploymentEvent => {
-  const { environment, deployable } = deployment;
+  try {
+    const { environment, deployable } = deployment;
 
-  return {
-    environment: {
-      category: mapEnvTierToCompassDeploymentEnv(environmentTier),
-      displayName: environment.name,
-      environmentId: environment.id.toString(),
-    },
-    pipeline: {
-      displayName: `${projectName} pipeline`,
-      pipelineId: deployable.pipeline.id.toString(),
+    const deploymentEvent = {
+      environment: {
+        category: mapEnvTierToCompassDeploymentEnv(environmentTier),
+        displayName: environment.name,
+        environmentId: environment.id.toString(),
+      },
+      pipeline: {
+        displayName: `${projectName} pipeline`,
+        pipelineId: deployable.pipeline.id.toString(),
+        url: deployable.pipeline.web_url,
+      },
+      sequenceNumber: deployment.id,
+      state: gitLabStateToCompassFormat(deployable.status.toUpperCase()),
+      description: `${projectName} deployment`,
+      displayName: `${projectName} deployment ${deployment.id}`,
+      lastUpdated: new Date(deployment.updated_at).toISOString(),
+      updateSequenceNumber: new Date(deployment.updated_at).getTime(),
       url: deployable.pipeline.web_url,
-    },
-    sequenceNumber: deployment.id,
-    state: gitLabStateToCompassFormat(deployable.status.toUpperCase()),
-    description: `${projectName} deployment`,
-    displayName: `${projectName} deployment ${deployment.id}`,
-    lastUpdated: new Date(deployment.updated_at).toISOString(),
-    updateSequenceNumber: new Date(deployment.updated_at).getTime(),
-    url: deployable.pipeline.web_url,
-  };
+    };
+    return deploymentEvent;
+  } catch (err) {
+    console.error(`There was an error mapping the gitlab API deployment: ${err.message}`);
```

Comment:
Is there any way we can keep track of these and then add a log at the end of the total number of deployments that couldn't be processed? This might help us when triaging customer issues to determine if it was a one-off or there's something consistently wrong with the shape of their data

Reply:
added!